lib/instrumentation/modules/pg.js

```diff
@@ -103,7 +103,15 @@ function patchClient (Client, klass, agent, enabled) {
             query.on('end', end)
             query.on('error', end)
           } else if (typeof query.then === 'function') {
-            query.then(end)
+            query = query
+              .then(function (result) {
+                end()
+                return result
+              })
+              .catch(function (err) {
+                end()
+                throw err
+              })
           } else {
```

Comment:
Prior to this PR this branch of the conditional would add our "end the span" callback to the end of the promise chain to ensure it was called.  I see that this PR expands on that to make sure any unhandled promise rejections are caught (avoid a fatal error) and then the error is re-thrown to ensure legitimate errors are now swallowed up.